### ISSUE TITLE
Fix macOS runtime issues

### ISSUE THREAD

**Issue by McCallisterRomer** [NcStudios/NcEngine#397]
Probably depends on #380

The sample does not run on macOS. Just tracking known issues here - maybe split them up if they require bigger fixes.
- Vulkan WIDE_LINES feature isn't available.
- Out of pool memory thrown when allocating first descriptor set.

**Comment by McCallisterRomer** [comment]
On out of pool memory issue, validation layers suggest its something else:
```
VUID-VkDeviceCreateInfo-pProperties-04451(ERROR / SPEC): msgNum: 976972960 - Validation Error: [ VUID-VkDeviceCreateInfo-pProperties-04451 ] Object 0: handle = 0x101fd7880, type = VK_OBJECT_TYPE_PHYSICAL_DEVICE; | MessageID = 0x3a3b6ca0 | vkCreateDevice: VK_KHR_portability_subset must be enabled because physical device VkPhysicalDevice 0x101fd7880[] supports it The Vulkan spec states: If the VK_KHR_portability_subset extension is included in pProperties of vkEnumerateDeviceExtensionProperties, ppEnabledExtensions must include "VK_KHR_portability_subset" (https://vulkan.lunarg.com/doc/view/1.2.198.1/mac/1.2-extensions/vkspec.html#VUID-VkDeviceCreateInfo-pProperties-04451)
    Objects: 1
        [0] 0x101fd7880, type: 2, name: NULL
enter CreateRenderingDescriptorPool
exit CreateRenderingDesciptorPool
VUID-VkDescriptorSetLayoutCreateInfo-pNext-pNext(ERROR / SPEC): msgNum: 729054019 - Validation Error: [ VUID-VkDescriptorSetLayoutCreateInfo-pNext-pNext ] Object 0: handle = 0x1200c2200, type = VK_OBJECT_TYPE_INSTANCE; | MessageID = 0x2b747b43 | vkCreateDescriptorSetLayout: Includes a pNext pointer (pCreateInfo->pNext) to a VkStructureType (VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_BINDING_FLAGS_CREATE_INFO), but its parent extension VK_EXT_descriptor_indexing has not been enabled. The Vulkan spec states: Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkDescriptorSetLayoutBindingFlagsCreateInfo or VkMutableDescriptorTypeCreateInfoVALVE (https://vulkan.lunarg.com/doc/view/1.2.198.1/mac/1.2-extensions/vkspec.html#VUID-VkDescriptorSetLayoutCreateInfo-pNext-pNext)
    Objects: 1
        [0] 0x1200c2200, type: 1, name: NULL
vk::Device::allocateDescriptorSetsUnique: ErrorOutOfPoolMemory%    
```




See here for info on setting portability bit (note this may require a later version of vulkan than the vma hpp supports):
https://vulkan-tutorial.com/Drawing_a_triangle/Setup/Instance#:~:text=is%20created%20successfully.-,Encountered%20VK_ERROR_INCOMPATIBLE_DRIVER,-%3A

**Comment by McCallisterRomer** [comment]
Also, wide lines may be a later vulkan feature, which could be resolved by moving to 1.3.xxx.

**Comment by McCallisterRomer** [comment]
Closing - macOS support no longer planned
